x/mongo/driver/topology/topology.go

```diff
@@ -69,6 +69,63 @@ const (
 	SingleMode
 )
 
+type hostEnv int
+
+const (
+	genuine hostEnv = iota
+	cosmosDB
+	doumentDB
+)
+
+var (
+	cosmosDBLog   = `You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb`
+	documentDBLog = `You appear to be connected to a DocumentDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/documentdb`
+)
+
+type envMap struct {
+	host string
+	env  hostEnv
+}
+
+type hostLogger struct {
+	logger *logger.Logger
+	envs   []envMap
+	logs   map[hostEnv]*string
+}
+
+func newHostLogger(l *logger.Logger) *hostLogger {
+	return &hostLogger{
+		logger: l,
+		envs: []envMap{
+			{".cosmos.azure.com", cosmosDB},
+			{".docdb.amazonaws.com", doumentDB},
+			{".docdb-elastic.amazonaws.com", doumentDB},
+		},
+		logs: map[hostEnv]*string{
+			cosmosDB:  &cosmosDBLog,
+			doumentDB: &documentDBLog,
+		},
+	}
+}
+
+func (l *hostLogger) log(host string) {
```

Comment:
Whether we decided to use a struct or not, we should use re-organize the code to match the existing "logTopology*" pattern:

```go
func logTopologyWrongServer(topo *Topology, host string) {
	envVars := []envMap{
		{".cosmos.azure.com", cosmosDB},
		{".docdb.amazonaws.com", doumentDB},
		{".docdb-elastic.amazonaws.com", doumentDB},
	}

	serverLogs := map[hostEnv]string{
		cosmosDB:  cosmosDBLog,
		doumentDB: documentDBLog,
	}

	for _, em := range envVars {
		if !strings.HasSuffix(host, em.host) {
			continue
		}

		if msg, ok := serverLogs[em.env]; ok {
			// Serialize the topology message to be consistent with other topology
			// messages.
			topo.cfg.logger.Print(logger.LevelInfo,
				logger.ComponentTopology,
				msg,
				logger.SerializeTopology(logger.Topology{ID: topo.id, Message: msg})...)
		}
	}
}
```